models/report.js

```diff
@@ -129,4 +147,65 @@ Report.findSortedPage = function(filter, page, callback) {
   });
 };
 
+Report.checkoutBatch = function(userId, callback) {
+  async.series([ 
+    this.releaseBatch, 
+    this.lockBatch.bind(this, userId, ITEMS_PER_BATCH),
+    this.loadBatch.bind(this, userId, ITEMS_PER_BATCH)
```

Comment:
@hooverlunch  lockBatch is an update query and it does not return updated records. I split them into two functions to keep them small but I can move the load batch under lock.